docs/developers/engineering-updates/2023-30-May-upgrades.md

```diff
@@ -0,0 +1,78 @@
+---
+title: 2023-30-05 Smart contract upgrades
+sidebar_position: 1
+---
+
+:::note
+
+Smart contract upgrades are being deployed to:
+
+- Linea devnet on May 30, 2023
+- Linea testnet on June 6, 2023
+
+This will cause some downtime. Continue reading to prepare your dapp for the new changes.
+
+:::
+
+## Context
+
+As part of the Alpha testnet upgrade, we are deploying new smart contracts (verifier, rollup, messaging service and token bridge).
+
+- There will be a new proxy/timelock setup, so the addresses will be different.
+- Storage/data layouts and the contract code is completely different, so we cannot keep the old addresses, as an upgrade will corrupt data.
+- The new contracts have the ability to Pause.
+- In the future, we will be using an upgrade pattern that does not cause downtime.
+
+## Order of operations
+
+1. Before the upgrade, finish all bridging transactions, and no new bridging transactions will be initiated.
+2. Stop the prover.
+3. Upgrade the system.
+4. Resume Layer 2.
+5. Resume prover.
+6. Resume bridging.
+
+## Key differences
+
+### ABI
+
+**Old ABI**
+
+```json
+abi
+```
+
+**New ABI**
+
+```json
+abi
+```
+
+### Input data
+
+**Old input data**
+
+**New input data**
+
+### Addresses
```

Comment:
Should I be adding all of this info for old contract addresses?
https://www.notion.so/consensys/Testnet-Runbook-Operation-a2574eced4f5442aa535cd65a44ab27d

How do we need to update for the new contract addresses immediately after ugprade?